cpp/src/io/comp/nvcomp_adapter.cpp

```diff
@@ -21,17 +21,29 @@
 
 #include <nvcomp/snappy.h>
 
+#define NVCOMP_DEFLATE_HEADER <nvcomp/deflate.h>
+#if __has_include(NVCOMP_DEFLATE_HEADER)
+#include NVCOMP_DEFLATE_HEADER
+#endif
+
 #define NVCOMP_ZSTD_HEADER <nvcomp/zstd.h>
 #if __has_include(NVCOMP_ZSTD_HEADER)
 #include NVCOMP_ZSTD_HEADER
-#define NVCOMP_HAS_ZSTD 1
+#endif
+
+#if NVCOMP_MAJOR_VERSION > 2 or (NVCOMP_MAJOR_VERSION == 2 and NVCOMP_MINOR_VERSION >= 3)
```

Comment:
I see we are dropping support for 2.1 and 2.2 here.  I am ok with this, assuming it is intentional.  For 22.10, we should only be pulling in 2.3 now.